wsimod/nodes/nodes.py

```diff
@@ -1142,6 +1173,19 @@ def __init__(self, decays={}, parent=None, **kwargs):
         self.end_timestep = self.end_timestep_decay
         self.ds = self.decay_ds
 
+    def apply_overrides(self, overrides: Dict[str, Any] = {}):
+        """Apply overrides to the decaytank.
+    
+        Enables a user to override any of the following parameters: 
+        decays.
+        
+        Args:
+            overrides (dict, optional): Dictionary of overrides. Defaults to {}.
+        """
+        self.decays = overrides.pop("decays", 
+                                    self.decays)
```

Comment:
I think we want to allow an override of an individual item in decay.

See [`process_parameters`](https://github.com/ImperialCollegeLondon/wsi/blob/5a267816ed4df5547071e2d4f09c6fe12d508ce8/wsimod/nodes/wtw.py#L147) for example

Reply:
I think this might link to my question here - https://github.com/ImperialCollegeLondon/wsi/pull/70/commits/7cfab64ec702a3542bc88744a2ff34093389edfb#r1509937857
Any thought here?

Reply:
Not sure I can see the question? It's just a commit at that link as far as I can tell

Reply:
Would you consider a situation where originally no process_parameters set for pollutant A but then the users want to append a set of process_parameters for the pollutant A? In this case, a Keyerror (no key found) will be raised.

Reply:
By append you mean provide in `overrides`?

Well currently `process_parameters` must either be provided in `config` or not at all and the defaults are used. We could define similar behaviour here with `overrides`, but I can't think of the use case. 

For `process_parameters` the intended use case of these `overrides` are, e.g., in an optimisation or sensitivity analysis we are introducing new or better technology to improve something - thus it will only impact one or few parameters.

For a `DecayTank` I think it is too general an object to think of such a clean use case - but certainly if this was used in a `Groundwater` node, you could envisage that you have some data on some pollutants, and so you include those in `overrides:decays` but not all?

Does it make sense? We can call about it if helpful!

Reply:
ok, I'm thinking for something like the `pollutant_load` in the `Surface`. Initially, there is no pollutant load ({}), but then the user would like to specify a {'nitrate': 5}. If we use `pollutant_load.update`, this may raise a keyerror? 
I guess a 'standard' rule is to only allow overrides that pre-exist. If the user would like to insert anything new, it should be pre-defined first?

Reply:
Would that crash it?

```
>>> a = {}
>>> a.update({'nitrate':1})
>>> a
{'nitrate': 1}
```

I think the point is that if we define behaviour for it in `apply_overrides`, then it can be overwritten by `overrides`. If we don't define behaviour for it, it will do nothing and print a warning:
https://github.com/ImperialCollegeLondon/wsi/blob/5a267816ed4df5547071e2d4f09c6fe12d508ce8/wsimod/nodes/nodes.py#L92

If we do not define behaviour for it, it can still 'manually' be changed, but may have unintended consequences if a user does not properly understand the subclass that they are changing.

Reply:
OK, though feels like the `self.decays.update(overrides.pop('decays', {}))` can do the job?